app/src/main/java/com/github/libretube/ui/adapters/CommentsPagingAdapter.kt

```diff
@@ -4,104 +4,78 @@ import android.view.LayoutInflater
-import com.github.libretube.ui.fragments.CommentsRepliesFragment
-import com.github.libretube.ui.viewholders.CommentsViewHolder
+import com.github.libretube.ui.viewholders.CommentViewHolder
 import com.github.libretube.util.HtmlParser
 import com.github.libretube.util.LinkHandler
 
-class CommentPagingAdapter(
-    private val fragment: Fragment?,
-    private val videoId: String,
+class CommentsPagingAdapter(
+    private val isReplies: Boolean,
     private val channelAvatar: String?,
-    private val isRepliesAdapter: Boolean = false,
-    private val handleLink: ((url: String) -> Unit)?,
-    private val dismiss: () -> Unit
-) : PagingDataAdapter<Comment, CommentsViewHolder>(DiffUtilItemCallback(
-    areItemsTheSame = { oldItem, newItem -> oldItem.commentId == newItem.commentId},
-    areContentsTheSame = { _, _ -> true },
-)) {
-    private var clickEventConsumedByLinkHandler = false
-
-    private fun navigateToReplies(comment: Comment) {
-        if (clickEventConsumedByLinkHandler) {
-            clickEventConsumedByLinkHandler = false
-            return
-        }
-
-        val args = bundleOf(
-            IntentData.videoId to videoId,
-            IntentData.comment to comment,
-            IntentData.channelAvatar to channelAvatar
-        )
-        fragment!!.parentFragmentManager.commit {
-            replace<CommentsRepliesFragment>(R.id.commentFragContainer, args = args)
-            addToBackStack(null)
-        }
+    private val handleLink: (url: String) -> Unit,
+    private val saveToClipboard: (Comment) -> Unit,
+    private val navigateToChannel: (Comment) -> Unit,
+    private val navigateToReplies: ((Comment, String?) -> Unit)? = null,
+) : PagingDataAdapter<Comment, CommentViewHolder>(
+    DiffUtilItemCallback(
+        areItemsTheSame = { oldItem, newItem -> oldItem.commentId == newItem.commentId },
+        areContentsTheSame = { _, _ -> true },
+    )
+) {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CommentViewHolder {
+        val layoutInflater = LayoutInflater.from(parent.context)
+        val binding = CommentsRowBinding.inflate(layoutInflater, parent, false)
+        return CommentViewHolder(binding)
     }
 
-    override fun onBindViewHolder(holder: CommentsViewHolder, position: Int) {
-        val comment = getItem(position)!!
-
+    override fun onBindViewHolder(holder: CommentViewHolder, position: Int) {
         holder.binding.apply {
+            val comment = getItem(position)!!
             commentAuthor.text = comment.author
             commentAuthor.setBackgroundResource(
                 if (comment.channelOwner) R.drawable.comment_channel_owner_bg else 0
             )
             commentInfos.text = comment.commentedTime
 
             commentText.movementMethod = LinkMovementMethodCompat.getInstance()
-            val linkHandler = LinkHandler {
-                clickEventConsumedByLinkHandler = true
-                handleLink?.invoke(it)
-            }
```

Comment:
This code existed for a reason.

When you click the link in the first comment of video `https://youtu.be/AdygBbbEnco` for example, the app opens the links **and** the comment's replies at the same time. However, when clicking a link, the replies should **not** be opened as a side effect.